library/alloc/src/collections/btree/map.rs

```diff
@@ -40,30 +40,15 @@ pub(super) const MIN_LEN: usize = node::MIN_LEN_AFTER_SPLIT;
 
 /// An ordered map based on a [B-Tree].
 ///
-/// B-Trees represent a fundamental compromise between cache-efficiency and actually minimizing
-/// the amount of work performed in a search. In theory, a binary search tree (BST) is the optimal
-/// choice for a sorted map, as a perfectly balanced BST performs the theoretical minimum amount of
-/// comparisons necessary to find an element (log<sub>2</sub>n). However, in practice the way this
-/// is done is *very* inefficient for modern computer architectures. In particular, every element
-/// is stored in its own individually heap-allocated node. This means that every single insertion
-/// triggers a heap-allocation, and every single comparison should be a cache-miss. Since these
-/// are both notably expensive things to do in practice, we are forced to, at the very least,
-/// reconsider the BST strategy.
+/// An ordered map is a map in which the keys are totally ordered.
```

Comment:
I've occasionally seen people be confused about ordered maps in this sense vs. "map that maintains *some consistent order*" (e.g., insertion order) as exemplified by Python's `OrderedDict` or http://crates.io/crates/ordermap in Rust. I'd throw in something about the entries being stored *in sorted order* before detailing what determines the sort order.

Reply:
This is a valuable insight and I can see how the first sentence can be mistaken for talking about a map with insertion order. It now says: "Given a key type with a [total order], an ordered map stores its entries in key order." which I feel is much harder to misunderstand. Let me know if you have more valuable suggestions.